lib/irc/Scheduler.js

```diff
@@ -0,0 +1,107 @@
+/*eslint no-invalid-this: 0*/
+"use strict"
+var promiseutil = require("../promiseutil");
+var Promise = require("bluebird");
+var logging = require("../logging");
+var log = logging.get("scheduler");
+
+function ScheduledQueue (server) {
```

Comment:
Oh and with a `killAll()` method, which can trivially be added to the existing `Queue`.